script/predicate_functions.sh

```diff
@@ -95,9 +95,22 @@ function is_ruby_25_plus {
   fi
 }
 
+function is_ruby_31_plus {
+  if ruby -e "exit(RUBY_VERSION.to_f >= 3.1)"; then
+    return 0
+  else
+    return 1
+  fi
+}
+
 function rspec_rails_compatible {
   if is_ruby_25_plus; then
-    return 0
+    # TODO remove when RSpec-Rails build is 3.1 safe by default
+    if is_ruby_31_plus; then
```

Comment:
[This](https://github.com/rspec/rspec-rails/pull/2552/files#diff-e07de5c8cc9353282e2b72f39e4baa0e3ac5f389d7578793910cc1e9a1fc6d2bR34) fixes the incompatibility. Do you want me to quickly extract and merge it?

Reply:
@JonRowe https://github.com/rspec/rspec-rails/pull/2553